1 - Introduction/JavaEndOfFile

```diff
@@ -0,0 +1,19 @@
+import java.io.*;
+import java.util.*;
+import java.text.*;
+import java.math.*;
+import java.util.regex.*;
+
+public class JavaEndOfFile {
+
+    public static void main(String[] args) {
+        Scanner scan = new Scanner(System.in);
+        int numLinea = 1;
+
+        while (scan.hasNext()==true){
```

Comment:
El `scan.hasNext()==true` se podría quedar simplemente en `scan.hasNext()` porque el método `hasNext()` ya te devuelve `true` o `false` según sea.

¿Podrías intentar implementar la misma solución pero con un `for` en vez de un `while`?
https://docs.oracle.com/javase/tutorial/java/nutsandbolts/for.html